SecureImage/UI/Albums/ImagePreviewTableViewCell.swift

```diff
@@ -0,0 +1,167 @@
+    }
+    
+    private func commonInit() {
+
+        let funccell = UINib(nibName: "AddImageCollectionViewCell" , bundle: nil)
+        collectionView.register(funccell, forCellWithReuseIdentifier: ImagePreviewTableViewCell.addImageCellReuseID)
+        
+        let imgcell = UINib(nibName: "PreviewImageCollectionViewCell" , bundle: nil)
+        collectionView.register(imgcell, forCellWithReuseIdentifier: ImagePreviewTableViewCell.imageThumbnailCellReuseID)
+
+        let layout = collectionView.collectionViewLayout as! UICollectionViewFlowLayout
+        layout.sectionInset = UIEdgeInsets(top: 0.0, left: ImagePreviewTableViewCell.leftInset,
+                                           bottom: ImagePreviewTableViewCell.bottomInset, right: ImagePreviewTableViewCell.rightInset)
+
+        collectionView.dataSource = self
+        collectionView.delegate = self
+    }
+
+    private func configureCell(cell: UICollectionViewCell, atIndexPath indexPath: IndexPath) {
+        
+        guard let previewImages = previewImages else {
+            fatalError("Unable configure collection view cell")
+        }
+        
+        switch indexPath.row {
+        case 0:
+            ()
+        default:
+            let document = previewImages[indexPath.row - 1]
+            (cell as! PreviewImageCollectionViewCell).document = document
+            (cell as! PreviewImageCollectionViewCell).onDeleteImageTouched = { (document: Document) in
+                print("I should delete \(document.id)")
+            }
+
+        }
+
+    }
+}
+
+extension ImagePreviewTableViewCell: UICollectionViewDelegateFlowLayout {
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+
+        let width = ImagePreviewTableViewCell.collectionViewRowHeightFor(collectionView.frame.size.width)
+
+        return CGSize(width: width, height: width)
+    }
+
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        
+        return ImagePreviewTableViewCell.cellRowSpacing
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        
+        return ImagePreviewTableViewCell.cellColumnSpacing
+    }
+}
+
+extension ImagePreviewTableViewCell: UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+
+        let count = previewImages?.count ?? 0
+        return count > 5 ? 6 : count + 1 // first 5images and one additional functional cell
```

Comment:
This returns up to count + 1 up to 6. this intended? max item# is 6?

Reply:
Yup, intended.